apiserver/facades/controller/undertaker/undertaker.go

```diff
@@ -54,13 +58,45 @@ func newUndertakerAPI(
 	return &UndertakerAPI{
 		st:                   st,
 		resources:            resources,
+		modelUUID:            modelUUID,
 		secretBackendService: secretBackendService,
 		modelInfoService:     modelInfoService,
+		cloudSpecGetter:      cloudSpecGetter,
 		ModelConfigWatcher:   commonmodel.NewModelConfigWatcher(modelConfigService, watcherRegistry),
-		CloudSpecer:          cloudSpecer,
 	}, nil
 }
 
+// CloudSpec returns the cloud spec used by the specified models.
+func (u *UndertakerAPI) CloudSpec(ctx context.Context, args params.Entities) (params.CloudSpecResults, error) {
+	results := params.CloudSpecResults{
+		Results: make([]params.CloudSpecResult, len(args.Entities)),
+	}
+	for i, arg := range args.Entities {
+		tag, err := names.ParseModelTag(arg.Tag)
+		if err != nil {
+			results.Results[i].Error = apiservererrors.ServerError(err)
+			continue
+		}
+		if tag.Id() != u.modelUUID.String() {
+			results.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrPerm)
+			continue
+		}
+		spec, err := u.cloudSpecGetter.GetCloudSpec(ctx)
+		if err != nil {
+			results.Results[i].Error = apiservererrors.ServerError(err)
+			continue
+		}
+		result := params.CloudSpecResult{
+			Error: apiservererrors.ServerError(err),
+		}
+		if err == nil {
```

Comment:
Small nit but it feels more idiomatic to put the if statement around the err case here.